app/bundles/EmailBundle/Assets/js/email.js

```diff
@@ -3,6 +3,12 @@ Mautic.emailOnLoad = function (container, response) {
     Mautic.internalDynamicContentItemCreateListeners = [];
     Mautic.internalDynamicContentFilterCreateListeners = [];
 
+    var urlParams = new URLSearchParams(window.location.search);
+    var emailType = urlParams.get('emailType');
+    if (emailType && typeof Mautic.selectEmailType === 'function') {
```

Comment:
Not sure why you're checking for this to be a function. Is the function just not available the first time this file is read, because that function doesn't exist yet? In that case, I'd suggest moving the `EmailOnLoad` event so it goes AFTER `selectEmailType`.

Reply:
We probably need this to happen with the EmailOnLoad because the user prompting for which resource type they want happens on EmailOnLoad
I believe that if we move the snippet location, we may have a delay (showing the type selection panel, even if for milisseconds) or the code being executed before the availability of the type choices (then basically losing the entire idea of this functionality)

What do you think?

Reply:
I think you misunderstand. I'm not suggesting moving the snippet, just moving the order of the functions in `email.js` and in `form.js` if `selectEmailType` is giving you trouble loading. Instead of doing a `typeof` check. But maybe you are doing that for another reason?

Eg. this is the order at the moment, in `email.js`:
- Mautic.emailOnLoad
- <bunch of other functions>
- Mautic.emailSendOnLoad 

The new order could be:
- Mautic.emailSendOnLoad 
- <bunch of other functions>
- Mautic.emailOnLoad